scripts/virtualized_install.sh

```diff
@@ -0,0 +1,481 @@
+    [[ -z "$internal_api_proxy_port" ]] && internal_api_proxy_port="8443"
+    [[ -z "$distro" ]] && distro="debian/11"
+    [[ -z "$proxy_url" ]] && proxy_url="127.0.0.1"
+    [[ -z "$app_port" ]] && app_port=$(generate_random_available_port)
+    [[ -z "$api_port" ]] && api_port=$(generate_random_available_port)
+    [[ -z "$env" ]] && env="production"
+    [[ -z "$show_in_console" ]] && show_in_console="true"
+    if [[ -z "$app_domain" ]]; then
+        app_domain=$(generate_random_string 10).nixopus.com
+    fi
+    if [[ -z "$api_domain" ]]; then
+        api_domain=$(generate_random_string 10).nixopus.com
+    fi
+    
+    local container_name=$(generate_container_name)
+    local app_proxy_name="app-proxy-${container_name}"
+    local api_proxy_name="api-proxy-${container_name}"
+    
+    CONFIG=(
+        ["distro"]="$distro"
+        ["api_domain"]="${api_domain:-}"
+        ["app_domain"]="${app_domain:-}"
+        ["proxy_url"]="$proxy_url"
+        ["app_port"]="$app_port"
+        ["api_port"]="$api_port"
+        ["email"]="$email"
+        ["password"]="$password"
+        ["env"]="$env"
+        ["container_name"]="$container_name"
+        ["base_distro"]=""
+        ["show_in_console"]="$show_in_console"
+        ["internal_app_proxy_port"]="$internal_app_proxy_port"
+        ["internal_api_proxy_port"]="$internal_api_proxy_port"
+        ["app_proxy_name"]="$app_proxy_name"
+        ["api_proxy_name"]="$api_proxy_name"
+    )
+    
+    CONFIG["base_distro"]=$(echo "${CONFIG[distro]}" | cut -d'/' -f1)
+}
+
+# Set up proxy for a specific service (app or api)
+function set_proxy_for_lxd() {
+    local service_type="$1"  # "app" or "api"
+    local container_name="${CONFIG[container_name]}"
+    local proxy_url="${CONFIG[proxy_url]}"
+    
+    # Determine configuration based on service type
+    if [[ "$service_type" == "app" ]]; then
+        local proxy_internal_port="${CONFIG[internal_app_proxy_port]}"
+        local proxy_external_port="${CONFIG[app_port]}"
+        local proxy_name="${CONFIG[app_proxy_name]}"
+    elif [[ "$service_type" == "api" ]]; then
+        local proxy_internal_port="${CONFIG[internal_api_proxy_port]}"
+        local proxy_external_port="${CONFIG[api_port]}"
+        local proxy_name="${CONFIG[api_proxy_name]}"
+    else
+        log_message "ERROR" "Invalid service type: $service_type. Must be 'app' or 'api'"
+        return 1
+    fi
+    
+    log_message "INFO" "Setting up ${service_type^^} proxy for container: $container_name"
+    log_message "INFO" "External port: $proxy_external_port -> Internal: $proxy_url:$proxy_internal_port"
+    
+    sudo lxc config device add "$container_name" "$proxy_name" proxy listen=tcp:0.0.0.0:"$proxy_external_port" connect=tcp:"$proxy_url":"$proxy_internal_port"
+}
+
+# Set up both app and API proxies
+function setup_all_proxies() {
+    set_proxy_for_lxd "app"
+    set_proxy_for_lxd "api"
+}
+
```

Comment:
Should we extract is out as constants at global level? Since i see multiple usage + in couple of places I see different cases for these as well.

I think would be better to keep it as constant and use where ever required directly?